src/listen/live.rs

```diff
@@ -38,46 +44,18 @@ where
     S: Stream<Item = std::result::Result<Bytes, E>>,
 {
     config: &'a Deepgram,
+    options: Option<&'a Options>,
     source: Option<S>,
-    encoding: Option<String>,
+    encoding: Option<Encoding>,
     sample_rate: Option<u32>,
     channels: Option<u16>,
+    endpointing: Option<Endpointing>,
+    utterance_end_ms: Option<u16>,
```

Comment:
Can we do a custom enum for `utterance_end_ms`, similar to endpointing? We'll want to enforce some guidance/limitation around possible values, so I think this makes sense.